### PR TITLE
physics and rigidbody done, no tests

### PR DESCRIPTION
**Pull request by adededede**
seems ok?

### CODE REVIEW

File: src/Physics.h [hunk 1]
```diff
@@ -1,11 +1,29 @@
...
+    btDefaultCollisionConfiguration* collisionConfiguration;
+    btCollisionDispatcher* dispatcher;
+    btBroadphaseInterface* broadphase;
+    btSequentialImpulseConstraintSolver* solver;
+    btDiscreteDynamicsWorld* dynamicsWorld;
+    std::vector<btRigidBody*> rigidBodies;
+    std::vector<btSoftBody*> softBodies;
 
-    void Update();
+    void update(int timeSinceLastFrame);
```
Comment (vaneri-9) [inline src/Physics.h:23]: You should probably in GameEngine add a new field for the variable you use here and in the MainGameLoop update that variable every iteration (or just use a pointer).


Then calling Update() should be possible (without the usage of an input parameter)

File: src/Physics.cpp [hunk 2]
```diff
@@ -1,6 +1,100 @@
 #include "Physics.h"
 
-void Physics::Update() {
-    // TODO: Write update for physics
-    return;
+Physics::Physics(float gravityX, float gravityY, float gravityZ ) {
```
Comment (vaneri-9) [inline src/Physics.cpp:3]: For earths gravity what would be the defaults? (0,0,9.82)?

**Review by vaneri-9 — CHANGES_REQUESTED** [review]
It would be nice if you could write here how the physics system even works, what "hitboxes" does it use, what's the difference between soft and hard bodies (even though it seems that right now there's only hard bodies coded).

File: src/Physics.cpp (hunk 2 above)
Comment (adededede) [reply src/Physics.cpp:3]: nope I wrote it in the comment of the cpp file it's 0, -9.82, 0
Maybe I should change the name of the variable it's more the forces expressed on each axes

File: src/Physics.h (hunk 1 above)
Comment (adededede) [reply src/Physics.h:23]: okay i'll look into it but then the new field in GameEngine need to be public

**Comment by adededede** [comment]
I feel like I explained it within the commentary of the cpp file, but if they weren't enough I can still add some explanation here of course

**Pull request merged**